setup.py

```diff
@@ -0,0 +1,67 @@
+"""
+Copyright 2018 SuperDARN
+
+setup.py
+2018-11-05
+To setup pyDARN as a third party library. Include installing need libraries for
+running the files.
+
+author:
+Marina Schmidt
+"""
+
+from distutils.core import setup
+from setuptools import setup, find_packages
+from os import path
+import sys
+from subprocess import check_call
+from setuptools.command.install import install
+
+# TODO: currently not implemented due to some challenges
+# with C API and memory leaks.
+#rstmodule = Extension('dmap',
+#                      sources= ['dmap.c'])
+
+class initialize_submodules(install):
+    def run(self):
+        if path.exists('.git'):
+            check_call(['git', 'submodule', 'update', '--init', '--recursive'])
+        if self.old_and_unmanageable or self.single_version_externally_managed:
+            return install.run(self)
+        #install.run(self)
+        caller = sys._getframe(2)
+        caller_module = caller.f_globals.get('__name__','')
+        caller_name = caller.f_code.co_name
+        if caller_module != 'distutils.dist' or caller_name!='run_commands':
+            # We weren't called from the command line or setup(), so we
+            # should run in backward-compatibility mode to support bdist_*
+            # commands.
+            install.run(self)
```

Comment:
Same thing here as the comment for line 30. Changing these 2 lines makes our custom `run` method do the same thing as the `setuptools` version, except we have that nicely injected `git` command!